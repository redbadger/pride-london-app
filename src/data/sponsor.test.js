// @flow
import decodeSponsor from "./sponsor";

describe("Sponsor", () => {
  describe("decoder", () => {
    it("correctly decodes valid CMS sponsor", () => {
      const data: mixed = {
        fields: {
          sponsorName: { "en-GB": "sponsorName" },
          sponsorLogo: { "en-GB": { sys: { id: "2o2SZPgYl2ABCWu2MoK333" } } },
          sponsorUrl: { "en-GB": "sponsorUrl" },
          sponsorLevel: { "en-GB": "Headline" }
        },
        sys: {
          id: "3O3SZPgYl2MUEWu2MoK2oi",
          contentType: {
            sys: {
              id: "sponsor"
            }
          },
          revision: 1
        }
      };

      const decoded = decodeSponsor("en-GB")(data);
      expect(decoded.ok).toEqual(true);
      if (decoded.ok) {
        expect(decoded.value).toMatchSnapshot();
      }
    });

    it("fails if a property is missing", () => {
      const data: mixed = {
        fields: {},
        sys: {
          id: "3O3SZPgYl2MUEWu2MoK2oi",
          contentType: {
            sys: {
              id: "sponsor"
            }
          }
        }
      };

      const decoded = decodeSponsor("en-GB")(data);
      expect(decoded.ok).toEqual(false);
    });
  });
});
