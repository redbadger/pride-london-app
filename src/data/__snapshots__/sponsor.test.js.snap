// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sponsor decoder correctly decodes valid CMS sponsor 1`] = `
Object {
  "contentType": "sponsor",
  "fields": Object {
    "sponsorLevel": "Headline",
    "sponsorLogo": Object {
      "sys": Object {
        "id": "2o2SZPgYl2ABCWu2MoK333",
      },
    },
    "sponsorName": "sponsorName",
    "sponsorUrl": "sponsorUrl",
  },
  "id": "3O3SZPgYl2MUEWu2MoK2oi",
  "locale": "en-GB",
  "revision": 1,
}
`;
